FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift

```diff
@@ -181,34 +187,43 @@ extension EncryptedStorage {
     }
 
     func updateKeys(with primaryFingerprint: String, passphrase: String?) {
-        let keys = keysInfo()
-            .filter { $0.primaryFingerprint == primaryFingerprint }
+        let keys = storage.objects(KeyInfoRealmObject.self).where {
+            $0.primaryFingerprint == primaryFingerprint
+        }
 
         try! storage.write {
             keys.map { $0.passphrase = passphrase }
         }
     }
 
-    func keysInfo() -> [KeyInfoRealmObject] {
-        let result = storage.objects(KeyInfoRealmObject.self)
-        return Array(result)
+    func findKeyInfo(by email: String) -> [KeyInfo] {
+        return storage.objects(KeyInfoRealmObject.self).where({
+            $0.account == email
+        }).map(KeyInfo.init)
+    }
+
+    func findPrivateKey(by email: String) -> [String] {
+        return storage.objects(KeyInfoRealmObject.self).where({
+            $0.account == email
+        }).map(\.private)
     }
```

Comment:
This one is the exact one as above, except the map - duplicates functionality.

We should remove `findKeyInfo` to `getPrivateKeys(for email: String)` and then remove `findPrivateKey`.

Reply:
When we do `.map(\.private)` Realm reads only this property from persistent storage, but `findKeyInfo` returns array of `KeyInfo` models with all properties.

Reply:
Got it - now I understood the motivation. Still, I think the optimization is not worth another method. I'd rather less code and more clarity, especially since the overhead here is very small.